src/models/device.ts

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { DeviceVerification } from "../crypto/deviceinfo";
+
+// user-Id → device-Id → IDevice
+export type DeviceMap = Map<string, Map<string, Device>>;
+
+type DeviceParameters = Pick<Device, "deviceId" | "userId" | "algorithms" | "keys"> & Partial<Device>;
+
+export class Device {
```

Comment:
also please can you make sure this and DeviceMap get exported via `matrix.ts` so that they appear in the main bit of the typedoc (currently they are marked as "internal": https://pr3272--js-sdk-docs-previews.netlify.app/stable/classes/_internal_.device)